src/rpc/net.cpp

```diff
@@ -200,9 +200,7 @@ static RPCHelpMan getpeerinfo()
         ServiceFlags services{fStateStats ? statestats.their_services : ServiceFlags::NODE_NONE};
         obj.pushKV("services", strprintf("%016x", services));
         obj.pushKV("servicesnames", GetServicesNames(services));
-        if (fStateStats) {
-            obj.pushKV("relaytxes", statestats.m_relay_txs);
-        }
+        obj.pushKV("relaytxes", fStateStats ? statestats.m_relay_txs : false);
```

Comment:
But this is incorrect. The default doesn't matter here - the value that it had for the entire connection is what it should be during disconnection. That value is gone at this point. So either we can't send the field, or can't send the peer entirely. Sending "false" is wrong.

Reply:
I think agree with the statement above, [fStateStats](https://github.com/jonatack/bitcoin/blob/2022-11-getpeerinfo-relayxes/src/net_processing.cpp#L1552) is only true if the peer id exists in [m_node_states](https://github.com/jonatack/bitcoin/blob/2022-11-getpeerinfo-relayxes/src/net_processing.cpp#L733) and [m_peer_map](https://github.com/jonatack/bitcoin/blob/2022-11-getpeerinfo-relayxes/src/net_processing.cpp#L730), so it seems like the node shouldn't exist in the return value at all.

I guess according to the comment at https://github.com/bitcoin/bitcoin/pull/26457#pullrequestreview-1181641835, there is the potential for a race condition, but seems awkward to replicate in the wild as any disconnection removes the nodes pointers and states entirely by the time the rpc is called.